### ISSUE TITLE
It seems that the GCN loss is not updating the model parameters.

### ISSUE THREAD

**Issue by lianshan01** [dinobby/MAGDi#2]
gcn_output, logits = self.gcn(graph_batch.x, graph_batch.edge_index)
        graph_batch.y = graph_batch.y.to(logits.device)
        node_loss = ce_cri(logits, graph_batch.y)

The input of the GCN loss seems to be independent of the model.

**Comment by dinobby** [comment]
Hi @lianshan01, thanks for your comment!
The input of GCN is the node embeddings obtained from the Multi-Agent Interaction Graphs (MAGs) using the same underlying decoder (Mistral), and we are updating the model through the sum of three objectives, i.e., the model is doing multi-task learning. Hope this clarifies your question!
